src/main/src/main/java/org/geoserver/catalog/StyleHandler.java

```diff
@@ -84,6 +84,20 @@ public String getFileExtension() {
     public String getCodeMirrorEditMode() {
         return getFormat();
     }
+    
+    /**
+     * Generates a template style using provided substitutionMap from StyleGenerator.
+     *
+     * @param type: the template type, see {@link org.geoserver.catalog.StyleType}.
+     * @param color: java.aw.Color to use during substitution
+     * @param colorName: Human readable color name, for use generating comments
+     * @param layerName: Layer name, for use generating comments
+     * 
+     * @return The text content of the style template after performing substitutions
+     */
+    public String getTemplate(StyleType type, Color color, String colorName, String layerName) {
```

Comment:
Consider renaming - now that are passing substitutions in this method produces a style definition (rather than a template).

Good idea changing from a Map<String,Object> to parameters, will force us to extend/document in the future if additional variations are allowed.